### DISCUSSION TITLE
Reverse relationship with adjacency model

### DISCUSSION THREAD

**Discussion by denizgolbas** [staudenmeir/laravel-adjacency-list#215] [Q&A] [answered]
I have recursive User model and Post model.

Post -> belongs to ->  User 

Is there a way to make this relation: PostUserAncestors ?

**Comment by denizgolbas** [comment]
When I try with hasManyDeepFromRelationsWithConstraints i get:

RuntimeException  Ancestors can only be at the beginning of deep relationships at the moment.

**Comment by staudenmeir** [comment, marked as answer]
Hi @denizgolbas,
It's not possible to define a relationship like this at the moment.

You can use this query as a workaround (does not work on MariaDB):

```php
class User extends Model
{
    public function recursivePosts()
    {
        return $this->hasManyOfDescendantsAndSelf(Post::class);
    }
}

$post = Post::find($id);

$users = User::whereHas('recursivePosts', function ($query) use ($post) {
    $query->where('posts.id', $post->id);
})->get();
```
